homeassistant/components/raspihats/switch.py

```diff
@@ -150,7 +135,12 @@ def should_poll(self):
     @property
     def is_on(self):
         """Return true if device is on."""
-        return self._state != self._invert_logic
+        try:
+            state = self.I2C_HATS_MANAGER.read_dq(self._address, self._channel)
```

Comment:
I think the problem is that we don't read the manager state after turning on/off the switch or when adding the entity. This is done here in the `is_on` property now, but we moved it to just the callback. We can probably fix it by calling the online_callback explicitly when adding the entity and when switching on/off.

Side note: It's a bit weird that the manager doesn't call the callback by itself after a new state is written to the switch.